translate/storage/lisa.py

```diff
@@ -329,20 +333,50 @@ def serialize_hook(treestring):
     def serialize(self, out=None):
         """Converts to a string containing the file's XML"""
         root = self.document.getroot()
+        xml_declaration = ""
+
         if self.XMLdoublequotes:
             if self.XMLuppercaseEncoding:
-                out.write(b'<?xml version="1.0" encoding="UTF-8"?>\n')
+                xml_declaration = (
+                    '<?xml version="1.0" encoding="' + self.encoding.upper() + '"?>\n'
+                )
+            else:
+                xml_declaration = (
+                    '<?xml version="1.0" encoding="' + self.encoding.lower() + '"?>\n'
+                )
+        else:
+            if self.XMLuppercaseEncoding:
+                xml_declaration = (
+                    "<?xml version='1.0' encoding='" + self.encoding.upper() + "'?>\n"
+                )
             else:
-                out.write(b'<?xml version="1.0" encoding="utf-8"?>\n')
+                xml_declaration = (
+                    "<?xml version='1.0' encoding='" + self.encoding.lower() + "'?>\n"
+                )
```

Comment:
When doing all this manually, it would be better to set `quote` and `encoding` based on the attributes and then build the XML declaration just once:

```suggestion
        xml_declaration = f"<?xml version={quote}1.0{quote} encoding={quote}{encoding}{quote}?>\n"
```